apollo-backend/src/main/java/io/logz/apollo/controllers/BlockerDefinitionController.java

```diff
@@ -188,9 +192,18 @@ public void updateBlockerDefinition(int id, String name, Integer environmentId,
         blockerDefinition.setStackId(stackId);
         blockerDefinition.setAvailability(availability);
         blockerDefinition.setBlockerTypeName(blockerTypeName);
-        blockerDefinition.setBlockerJsonConfiguration(blockerJsonConfiguration);
         blockerDefinition.setActive(isActive);
 
+        if (blockerTypeName.equals(BlockerTypeName.SINGLE_REGION)) {
+            blockerJsonConfiguration = initSingleRegionBlockerJsonConfig(environmentId, serviceId, stackId, availability, blockerJsonConfiguration, req);
+            if (blockerJsonConfiguration == null) return;
```

Comment:
i think you need to handle the response message before return

Reply:
@matvey-mtn The response message is done inside of the init

Reply:
@matvey-mtn I can override the request response variables values by explicit make another assignment in the update endpoint method.

Reply:
@maorshap either throw exception in initSingleRegionBlockerJsonConfig and handle in controller (with assignment) or update code as you suggested - explicitly assign code & response message
In the second case need to check that override works

Reply:
@matvey-mtn Going on the second option then

Reply:
you still need to set blocker configuration in else

```
else {
   blockerDefinition.setBlockerJsonConfiguration(blockerJsonConfiguration);
  }
```

Reply:
@matvey-mtn done.